### ISSUE TITLE
Parser explicitly reports wrong element name in stack

### ISSUE THREAD

**Issue by Worthaboutapig** [ractivejs/ractive#2898]
## Description:
When a ParseError is thrown and the error relates to a tag/element, e.g. a missing closing '/' (see fiddle), then the error message is taken from the `elementStack`, in which the element name is explicitly lower-cased `var lowerCaseName = element.e.toLowerCase();` and will report an incorrect name for the element/tag that's missing the end '/' (i.e. it reports 'component', instead of 'Component'.

Are element names case-sensitive?... this implies they're not, but either way, it's misleading. Minor, in a sense, but errors with misleading/ wrong info is pretty bad, IMHO.

## Versions affected:
0.8.11

## Platforms affected:
All

## Reproduction:
https://jsfiddle.net/worthaboutapig/qzykkjef/

## Code:

**Comment by ghost** [comment]
It might be related to an earlier issue https://github.com/ractivejs/ractive/issues/2821

Also, your fiddle appears to work. I see no errors in the console. Did you forget to break the code? 😄

**Comment by evs-chris** [comment]
No, case doesn't matter for elements. Case _does_ matter for components though, and I'm not entirely sure that it should. The reason the tag names are lower cased in the tag stack is that this is technically valid html `<a ...>why?</A>`. Components, which look like elements until runtime, are case sensitive though, so the original casing on the opening tag is kept around so that the appropriate component can be looked up. You could argue that _all_ tags should be case insensitive, and I would agree because it would be bloody confusing for `<MyComponent />` and `<myComponent />` to be two different things.

**Comment by Worthaboutapig** [comment]
If components are case-sensitive, then the stack trace is wrong, as the component tag name is always lower-cased.  I've updated the fiddle to demonstrate; firstly that having two components differing only in case returns a stack-trace with identical errors and that the case of the tag name is ignored.
(you need to open the console debugger to see the error output).

So, should this be fixed, or should component names be case-insensitive?

**Comment by MartinKolarik** [comment]
>You could argue that all tags should be case insensitive, and I would agree because it would be bloody confusing for <MyComponent /> and <myComponent /> to be two different things.

I was thinking about making components case insensitive in the past so that we could lowercase everything during parse time but decided it was probably better not to do so. I think the original idea behind case sensitive components was that you can have components with the same name as regular html tags without completely overriding them. E.g., having a `<Video>` component as a custom wrapper around native `<video>`.
